x-pack/test/api_integration/apis/ml/data_frame_analytics/update.ts

```diff
@@ -0,0 +1,236 @@
+          .post(`/api/ml/data_frame/analytics/${analyticsId}/_update`)
+          .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+          .set(COMMON_REQUEST_HEADERS)
+          .send(requestBody)
+          .expect(200);
+
+        expect(body.allow_lazy_start).to.eql(requestBody.allow_lazy_start);
+        expect(body.description).to.eql(testJobConfigs[2].description);
+        expect(body.model_memory_limit).to.eql(commonJobConfig.model_memory_limit);
+        expect(body.max_num_threads).to.eql(commonJobConfig.max_num_threads);
+      });
+
+      it('should only update model_memory_limit field of analytics job when model_memory_limit is sent in request', async () => {
+        const analyticsId = `${jobId}_3`;
+
+        const requestBody = {
+          model_memory_limit: '61mb',
+        };
+
+        const { body } = await supertest
+          .post(`/api/ml/data_frame/analytics/${analyticsId}/_update`)
+          .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+          .set(COMMON_REQUEST_HEADERS)
+          .send(requestBody)
+          .expect(200);
+
+        expect(body.model_memory_limit).to.eql(requestBody.model_memory_limit);
+        expect(body.allow_lazy_start).to.eql(commonJobConfig.allow_lazy_start);
+        expect(body.description).to.eql(testJobConfigs[3].description);
+        expect(body.max_num_threads).to.eql(commonJobConfig.max_num_threads);
+      });
+
+      it('should only update max_num_threads field of analytics job when max_num_threads is sent in request', async () => {
+        const analyticsId = `${jobId}_4`;
+
+        const requestBody = {
+          max_num_threads: 2,
+        };
+
+        const { body } = await supertest
+          .post(`/api/ml/data_frame/analytics/${analyticsId}/_update`)
+          .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+          .set(COMMON_REQUEST_HEADERS)
+          .send(requestBody)
+          .expect(200);
+
+        expect(body.max_num_threads).to.eql(requestBody.max_num_threads);
+        expect(body.model_memory_limit).to.eql(commonJobConfig.model_memory_limit);
+        expect(body.allow_lazy_start).to.eql(commonJobConfig.allow_lazy_start);
+        expect(body.description).to.eql(testJobConfigs[4].description);
+      });
+
+      it('should not allow to update analytics job for unauthorized user', async () => {
+        const analyticsId = `${jobId}_0`;
+        const requestBody = {
+          description: 'Unauthorized',
+        };
+
+        const { body } = await supertest
+          .post(`/api/ml/data_frame/analytics/${analyticsId}/_update`)
+          .auth(USER.ML_UNAUTHORIZED, ml.securityCommon.getPasswordForUser(USER.ML_UNAUTHORIZED))
+          .set(COMMON_REQUEST_HEADERS)
+          .send(requestBody)
+          .expect(404);
+
+        expect(body.error).to.eql('Not Found');
+        expect(body.message).to.eql('Not Found');
```

Comment:
Would be good to validate that the description hasn't changed.

Reply:
Updated in defc6bd35f70cccb76decd7443ee22375902978f